src/main/scala/com/fulcrumgenomics/umi/ReviewConsensusVariants.scala

```diff
@@ -52,82 +129,197 @@ import scala.collection.mutable.ListBuffer
+    val intervals = new IntervalList(consensusReader.getFileHeader)
+    variants.foreach (v => intervals.add(v.toInterval))
+
+    // Make locus iterators for both the consensus and grouped BAMs
+    val Seq(consensusIterator, groupedIterator) = Seq(consensusReader, groupedReader).map { r =>
+      val iterator = new SamLocusIterator(r, intervals, true)
+      iterator.setEmitUncoveredLoci(true)
+      iterator.setMappingQualityScoreCutoff(0)
+      iterator.setQualityScoreCutoff(0)
+      iterator.setSamFilters(Collections.emptyList())
+      iterator.setIncludeIndels(true)
+      iterator.iterator()
+    }
+
+    val writer = Metric.writer[ConsensusVariantReviewInfo](output)
+
+    for (((variant, consensusPileup), groupedPileup) <- variants.iterator.zip(consensusIterator).zip(groupedIterator)) {
+      require(variant.chrom == consensusPileup.getSequenceName, "Variants and consensus iterator out of sync.")
+      require(variant.chrom == groupedPileup.getSequenceName,   "Variants and grouped iterator out of sync.")
+      require(variant.start == consensusPileup.getPosition,     "Variants and consensus iterator out of sync.")
+      require(variant.start == groupedPileup.getPosition,       "Variants and grouped iterator out of sync.")
+
+      // Setup ready to iterate through the consensus reads
+      val consensusCounts  = BaseCounts(consensusPileup.getRecordAndOffsets.asScala)
+
+      val rawByMiAndReadNum = groupedPileup.getRecordAndOffsets.asScala.groupBy(r => toMi(r.getRecord) + readNumberSuffix(r.getRecord))
+
+      consensusPileup.getRecordAndOffsets
+        .filter(r => r.getReadBase.toChar.toUpper != variant.refBase)
+        .filterNot(r => this.ignoreNsInConsensusReads && r.getReadBase.toChar.toUpper == 'N')
+        .toSeq.sortBy(r => toMi(r.getRecord) + readNumberSuffix(r.getRecord))
+        .foreach { c =>
+          val mi = toMi(c.getRecord)
+          val consensusReadName = c.getRecord.getReadName + ( if (c.getRecord.getReadPairedFlag && c.getRecord.getSecondOfPairFlag) "/2" else "/1" )
+          val rawCounts = BaseCounts(rawByMiAndReadNum(mi + readNumberSuffix(c.getRecord)))
+
+          val m =  ConsensusVariantReviewInfo(
+            chrom = variant.chrom,
+            pos   = variant.start,
+            ref   = variant.refBase.toString,
+            genotype = variant.genotype.getOrElse("NA"),
+            filters  = variant.filters.getOrElse(""),
+            A = consensusCounts.a,
+            C = consensusCounts.c,
+            G = consensusCounts.g,
+            T = consensusCounts.t,
+            N = consensusCounts.n,
+            consensus_read = consensusReadName,
+            consensus_insert = toInsertString(c.getRecord),
+            consensus_call = c.getReadBase.toString.toUpperCase,
+            consensus_qual = c.getBaseQuality,
+            a = rawCounts.a,
+            c = rawCounts.c,
+            g = rawCounts.g,
+            t = rawCounts.t,
+            n = rawCounts.n
```

Comment:
I don't really see the point ... that's actually _more_ code since this is the only use of the constructor.  So I'd have to have a second apply() method that does this, but still the main constructor/apply to keep Metric happy.